clang/docs/SYCLFERFCLoweringKernel.md

```diff
@@ -0,0 +1,84 @@
+# SYCL Kernel
+
+A SYCL construct such as `parallel_for` or 'single_task' takes a named function object or a lambda as one of its arguments.   The contents of this function object are executed on the device.   However, as the SYCL runtime can rely on other offload APIs like OpenCL or CUDA to execute the function object, it needs to respect the calling convention of these API.  To enable this, the function object is converted into the format of an OpenCL kernel.
+
+Consider the following code snippet:
+
+```
+  sycl::accessor<char, 1, sycl::access::mode::read> acc1;
+  int i = 13;
+  sycl::sampler smplr;
+
+  struct S {
+    char c;
+    int i;
+  } test_s;
+
+  test_s.c = 14;
+
+  kernel_single_task<class first_kernel>([=]() {
+    if (i == 13 && test_s.c == 14) {
+
+      acc1.use();
+      smplr.use();
+    }
+  });
+```
```

Comment:
Same comment here about using real SYCL code.  I think you can reuse the same example code I suggested in #11431.